File: Leetcode/Python/_134.py
class Solution:
    def canCompleteCircuit(self, gas: List[int], cost: List[int]) -> int:
        start_pos = 0
        surplus = 0
        deficit = 0
        for i in range(len(gas)):
            surplus += gas[i] - cost[i]
            if surplus < 0:
                start_pos = i + 1
                deficit += surplus
                surplus = 0
        return start_pos if surplus + deficit >= 0 else -1

File: Leetcode/cpp/134.cpp
class Solution {
public:
    int canCompleteCircuit(vector<int>& gas, vector<int>& cost) {
        int total = 0;
        int current = 0;
        int index = 0;
        for (int i = 0; i < gas.size(); ++i) {
            total += gas[i] - cost[i];
            current += gas[i] - cost[i];
            if (current < 0) {
                current = 0;
                index = i + 1;
            }
        }
        return total >= 0 ? index : -1;
    }
};
